src/pages/tutorials/build-an-nft-marketplace/part-2.md

```diff
@@ -603,238 +608,238 @@ You are the owner of this bottle so you can create an offer to sell it.
-                        </Button>
-                      </form>
-                    )}
-                  </CardActions>
-                </Card>
-              ))}
-          </ImageList>
-        </Fragment>
-      ) : (
-        <Typography sx={{ py: "2em" }} variant="h4">
-          Sorry, there is not NFT to buy yet, you need to mint or sell bottles
-          first
-        </Typography>
-      )}
-    </Paper>
-  );
-}
-```
+                      />
+                      <CardMedia
+                        sx={{ width: "auto", marginLeft: "33%" }}
+                        component="img"
+                        height="100px"
+                        image={nftContratTokenMetadataMap
+                          .get(token_id.toString())
+                          ?.thumbnailUri?.replace(
+                            "ipfs://",
+                            "https://gateway.pinata.cloud/ipfs/"
+                          )}
+                      />
+
+                      <CardContent>
+                        <Box>
+                          <Typography variant="body2">
+                            {" "}
+                            {"Price : " + offer.price.dividedBy(1000000) + " XTZ"}
+                          </Typography>
+                        </Box>
+                      </CardContent>
+
+                      <CardActions>
+                        {!userAddress ? (
+                          <Box marginLeft="5vw">
+                            <ConnectButton
+                              Tezos={Tezos}
+                              nftContratTokenMetadataMap={
+                                nftContratTokenMetadataMap
+                              }
+                              setUserAddress={setUserAddress}
+                              setUserBalance={setUserBalance}
+                              wallet={wallet}
+                            />
+                          </Box>
+                        ) : (
+                          <form
+                            style={{ width: "100%" }}
+                            onSubmit={(values) => {
+                              setSelectedOfferEntry([token_id, offer]);
+                              formik.handleSubmit(values);
+                            }}
+                          >
+                            <Button type="submit" aria-label="add to favorites">
+                              <ShoppingCartIcon /> BUY
+                            </Button>
+                          </form>
+                        )}
+                      </CardActions>
+                    </Card>
+                  ))}
+              </ImageList>
+            </Fragment>
+          ) : (
+            <Typography sx={{ py: "2em" }} variant="h4">
+              Sorry, there is not NFT to buy yet, you need to mint or sell bottles
+              first
+            </Typography>
+          )}
+        </Paper>
+      );
+    }
+    ```
 
 ## Buy some wine!
 
```

Comment:
"Now you can see on **Trading** menu the **Wine catalogue** submenu, click on it." => "In the **Wine catalogue** submenu, click **Trading**."